lib/rubocop/cop/lint/useless_default_value_argument.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for usage of method `fetch` or `Array.new` with default value argument
+      # and block. In such cases, block will always be used as default value.
+      #
+      # This cop emulates Ruby warning "block supersedes default value argument" which
+      # applies to `Array.new`, `Array#fetch`, `Hash#fetch`, `ENV.fetch` and
+      # `Thread#fetch`.
+      #
+      # @safety
+      #   This cop is unsafe because the receiver could have nonstandard implementation
+      #   of `fetch`, or be a class other than the one listed above.
```

Comment:
In the `Lint` department of static analysis, targeting variables carries a high risk of false positives. Receivers should likely be limited to specific, such as` Array.new`, `[].fetch`, `{}.fetch`, `ENV.fetch`, or `Thread.current.fetch`. And cases where the behavior of these standard APIs is overridden likely do not need to be considered in terms of safety.

On the other hand, autocorrection may be unsafe, as it could remove side effects like the following:

```console
$ ruby -we 'def x(a) = puts "side effect"; Array.new(5, x(1)) { 2 }'
side effect
-e:1: warning: block supersedes default value argument
```

Reply:
> On the other hand, autocorrection may be unsafe, as it could remove side effects like the following

👍 I've noted this in the docs.